### DISCUSSION TITLE
Defining a custom operator without needing the sympy mapping

### DISCUSSION THREAD

**Discussion by MilesCranmer** [MilesCranmer/PySR#233] [Show and tell]
This was in an email discussion with a user: 

> How can I define a custom operator, without needing to define a sympy mapping for it?

The solution is to define a `sympy.Function` with the name of your custom operator - then sympy won't map it into other sympy functions.

For example, say we want to define a custom logarithm function:

```python
import sympy
from pysr import PySRRegressor

class my_log(sympy.Function):
    pass

model = PySRRegressor(
    unary_operators=["my_log(x::T) where T = (x > zero(T)) ? log(x) : T(NaN)"], 
    extra_sympy_mappings={"my_log": my_log}
)
```

Then, this can be used normally, and the sympy export will just use `my_log`, rather than needing to convert it to native sympy functions.

This could also be used if you want to define custom LaTeX for some operator.

**Comment by AthulKunjipurayil** [comment]
model = PySRRegressor(
    niterations=5,
    binary_operators=["+", "*",],
    unary_operators=["exp", "sin", "myf(x)"],
    extra_sympy_mappings={ 
    'myf': lambda x: sympy.pow(x, 1/3)
    },
    **default_pysr_params,
)

model.fit(X, y)
Hi Miles,
1. I am trying to learn PySR and was trying to use pow(x, 1/3). I couldn't find out why it says "myf is not defined". I haven't used Juila before. 
2. Can I define functions involving (X_0+X_1)/X_2 as user defined operators here? 
Thanks!

**Reply by MilesCranmer** [reply]
See example on the readme- need to give a definition to myf, like "myf(x) = abs(x) ^ (1//3)"

**Reply by AthulKunjipurayil** [reply]
I looked at it and worked.
Thanks
